app/src/main/java/com/lagradost/cloudstream3/ui/account/AccountDialog.kt

```diff
@@ -48,11 +52,32 @@ object AccountDialog {
                 callback.invoke(null)
             }
             .setOnDismissListener {
-                if (binding.pinEditTextError.isVisible()) {
+                if (binding.pinEditTextError.isVisible) {
                     callback.invoke(null)
                 }
             }
-            .create()
+
+        if (isNewPin) {
+            if (errorText != null) binding.pinEditTextError.visibleWithText(errorText)
+            builder.setPositiveButton(R.string.setup_done) { _, _ ->
+                if (binding.pinEditTextError.isVisible) {
+                    // If the done button is pressed and there is an error,
+                    // ask again, and mention the error that caused this.
+                    showPinInputDialog(
+                        context = binding.root.context,
+                        currentPin = null,
+                        editAccount = true,
+                        errorText = binding.pinEditTextError.text.toString(),
+                        callback = callback
+                    )
```

Comment:
Just noting this basically is meant to give the same results as the keyboard done listener down below:
```kt
binding.pinEditText.setOnEditorActionListener { _, actionId, _ ->
            if (actionId == EditorInfo.IME_ACTION_DONE && isPinValid) {
                val enteredPin = binding.pinEditText.text.toString()
                callback.invoke(enteredPin)
                dialog.dismissSafe()
            }
            true
        }
        ```

But, we can't stop this from dismissing the dialog it seems, so we just ask again, until there is no error, or the user cancels it, and we make sure to show the error again so the user knows what went wrong. It just makes it easier so you don't have to toggle the checkbox again, or wonder went went wrong if you accidently press "Done" with no error displaying afterword's. It's not the best practice for this to call itself but it works, and just makes it require less manual work, making it easier on TV for example where this is kinda hard to navigate sometimes.